users/vieting/jobs/fairseq.py

```diff
@@ -204,4 +204,127 @@ def create_dict_ltr(self):
 """
         with open(self.out_dict_ltr_path.get(), 'w') as f:
             f.write(dict_ltr_content)
-    
\ No newline at end of file
+
+
+class FairseqDecodingJob(Job):
+    """
+    Runs decoding with fairseq on a given fine-tuned model and a given data set.
+
+    Yields hypotheses and targets files in both word format and specified post-processing format.
+    """
+    def __init__(
+            self,
+            fairseq_python_exe: tk.Path,
+            fairseq_root: tk.Path,
+            model_path: tk.Path,
+            data_path: tk.Path,
+            gen_subset: str,
```

Comment:
What does `"gen_"` mean?

Reply:
I assume `gen_` is an abbreviation for "generate" (or an associated word). This was also the parameter name used by fairseq, so for consistency reasons (and because I couldn't think of a better name), I also used this parameter name.

Reply:
Okay, I don't think it's very expressive but let's keep it like this then.